drivers/timer/stm32_lptim_timer.c

```diff
@@ -61,11 +64,63 @@ static const struct device *const clk_ctrl = DEVICE_DT_GET(STM32_CLOCK_CONTROL_N
  *    0xFFFF / (LSE freq (32768Hz) / 128)
  */
 
-static uint32_t lptim_clock_freq = KHZ(32);
-static int32_t lptim_time_base;
+#if (CONFIG_STM32_LPTIM_CLOCK_LSI)
+/*
+ * Most of the STM32_SERIES have a LSI @32000Hz,
+ * this frequency might differ on some that do not yet use LPTIM
+ */
+static uint32_t lptim_clock_freq = KHZ(32); /* Force STM32_SRC_LSI */
 
-/* The prescaler given by the DTS and to apply to the lptim_clock_freq */
-#define LPTIM_CLOCK_RATIO DT_PROP(DT_DRV_INST(0), st_prescaler)
+/* Kconfig defines the clock source as LSI : check coherency with DTS */
+#if (DT_INST_CLOCKS_CELL_BY_IDX(0, 1, bus) != STM32_SRC_LSI)
+#warning CONFIG_STM32_LPTIM_CLOCK_LSI requires STM32_SRC_LSI defined as LPTIM domain clock
+#endif /* STM32_SRC_LSI */
+
+/* Kconfig is setting CONFIG_SYS_CLOCK_TICKS_PER_SEC adapted to the clock ratio */
+#if !defined(CONFIG_STM32_LPTIM_TICK_FREQ_RATIO_OVERRIDE)
+/* For prescaler = 1,2,4,8 CONFIG_SYS_CLOCK_TICKS_PER_SEC is <= 4000 */
```

Comment:
/* For prescaler = 1,2,4,8 CONFIG_SYS_CLOCK_TICKS_PER_SEC <= 4000 is advised */

Reply:
comment changed